### PR TITLE
Fan Feng's assignment

### PR DESCRIPTION
**Pull request by Dante-Vonarmia**
Please let me know if you have any question and further requirement.
Link: https://github.com/Dante-Vonarmia/lightning-talk-polling-app

Featuring:
1. Extensible Front End structure consists of React / Redux / Redux-Saga.
2. Isolated RESTful API server
3. Passport Strategy
4. Model Validation Strategy
5. Flexible Webpack configuration
6. Clean code style
7. Debuggable

Existing Potential issue
1. CORS error in Client Side
2. Need CSS decoration
3. Adding Unit Testing
4. I wanted to add ranking algorithm in mongoose query, but this cost me too much of time.

### CODE REVIEW

File: client/config/webpack.dev.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          // Compiles Sass to CSS
+          "sass-loader",
+        ],
+      },
+    ]
+  },
+
+  plugins: [
+    // Only update what has changed on hot reload
+    new webpack.HotModuleReplacementPlugin(),
```
Comment (johncalvinroberts) [inline client/config/webpack.dev.js:41]: Have you thought about trying React Fast Refresh for hot module replacement?

File: client/src/views/components/PollsList/item.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const Item = ({ item }) => {
+  const dispatch = useDispatch();
+
+  return (
+    <li>
+      <button onClick={() => dispatch(pollActions.updatePollVote(item._id))}> Vote {item.votes} </button>
+      <h5>Topics Title</h5>
+      <p>
+        <span>By: Authors name</span>
+        <span>{moment(item.date_created).fromNow()}</span>
```
Comment (johncalvinroberts) [inline client/src/views/components/PollsList/item.js:16]: `moment` is being deprecated, I think you can try using a more [lightweight alternative](https://bundlephobia.com/result?p=moment@2.29.1)

File: client/src/views/pages/Auth.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+class Auth extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+  render() {
+    return (
+      <div>
+        <h1>Login</h1>
+        {<Login authUser={this.props.authUser} />}
```
Comment (johncalvinroberts) [inline client/src/views/pages/Auth.js:15]: Why do we need to wrap this component in curly braces?

File: client/src/views/pages/Home.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        <PollsList
+          polls={polls}
+          authed={this.props.authed}
+          updatePollVote={this.props.updatePollVote}
+        />
+      </div>
+    ) : (
+      <div>No content.</div>
+    )
+  }
```
Comment (johncalvinroberts) [inline client/src/views/pages/Home.js:21]: Could you try adding a `loading` state here?

**Review by johncalvinroberts — COMMENTED** [review]
Hi, Fan! I left a few comments. Feel free to update or respond to my comments with your reasoning.

File: client/config/webpack.dev.js (hunk 1 above)
Comment (Dante-Vonarmia) [reply client/config/webpack.dev.js:41]: Wow, that's cool! I'll definitely try it out.

File: client/src/views/components/PollsList/item.js (hunk 2 above)
Comment (Dante-Vonarmia) [reply client/src/views/components/PollsList/item.js:16]: So cool! It really helpful to improve my libs resource. Trying it...

File: client/src/views/pages/Auth.js (hunk 3 above)
Comment (Dante-Vonarmia) [reply client/src/views/pages/Auth.js:15]: Sorry. I had a CORS problem there. And I think I just canceled comment here. I just missed this place.
BTW, if I could solve axios local CORS conflict problem. This App could be run freely.

File: client/src/views/pages/Home.js (hunk 4 above)
Comment (Dante-Vonarmia) [reply client/src/views/pages/Home.js:21]: Yeah, for sure! I do have "loading" state in my reducer. Just like I said about the local CORS issue. My progress on the business layer was stucked for a very very long time. I'll improve these within this week.
Thank you so much for the suggestions.

**Comment by Dante-Vonarmia** [comment]
Issue 1 solved.
Issue 2 solved.

New issue Login Passport authentication no response.
Checking...

**Comment by makara** [comment]
This should be enough for us to make a decision. Closing for now. Thanks everyone here.

**Pull request closed without merging**